src/lib/raft/raft.c

```diff
@@ -723,13 +723,18 @@ raft_worker_handle_io(struct raft *raft)
 					raft_sm_become_leader(raft);
 				else
 					raft_sm_become_candidate(raft);
-			} else if (raft->vote != 0) {
+			} else if (raft->vote != 0 || raft->last_demoted_term ==
+						      raft->volatile_term) {
 				/*
-				 * Voted for some other node. Wait if it manages
-				 * to become a leader.
+				 * Voted for some other node or resigned from
+				 * leadership in the current turn (i.e., cannot
```

Comment:
typo: turn -> term?
Or maybe rephrase this somehow. "resigned from leadership in the just-ended term", for example.

Reply:
Dropped this hunk